documentation/DPG-quickstart.md

```diff
@@ -0,0 +1,298 @@
+
+For an overview of the review and release process for new libraries, visit: https://aka.ms/azsdk/dpcodegen.
+
+If you have code generation queries, post them in the [TypeSpec Discussion](https://teams.microsoft.com/l/channel/19%3a906c1efbbec54dc8949ac736633e6bdf%40thread.skype/TypeSpec%2520Discussion%2520%25F0%259F%2590%25AE?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) channel. Tag `@DPG TypeScript` for JavaScript/TypeScript-specific inquiries.
+
+Join the [JavaScript - Reviews](https://teams.microsoft.com/l/channel/19%3a408c5f1322ee4303b02b5da9c5ff6137%40thread.skype/Language%2520-%2520JavaScript%2520-%2520Reviews?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) channel for assistance with the API review process.
+
+# Sett up your development environment
+
+Follow the [setup guide](https://github.com/Azure/azure-sdk-for-js/blob/main/CONTRIBUTING.md#prerequisites) for environment prerequisites in the Azure SDK for JS repository.
+
+# Identify your project's service and package name
+
+The `service name` is a concise identifier for the Azure service and should be consistent across all SDK languages. It's typically the name of the directory in the azure-rest-api-specs repository containing your service's REST API definition.
+
+The `package name` is used when publishing to [npmjs](https://www.npmjs.com/). It usually follows the format `@azure/{service-name}` or `@azure/{service-name}-{module}` for services with multiple modules.
+
+# Structure your project
+
+1. **SDK Repo Root**: the generated libraries should be in the [azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js) repo, so fork and clone it in your local then the absolute path is called **${SDK_REPO_ROOT} folder**.
+
+1. **Project Folder Structure**: the typical structure is `sdk/{servicename}/{servicename}-{modulename}`, e.g., `sdk/storage/storage-blob`. That folder is under {SDK_REPO_ROOT} and will be your **${PROJECT_ROOT} folder**. 
+
+1. **Package Name Convention**: follow the format `@azure/{service-name}-{module}`, like `@azure/storage-blob`.
+
+
+# Steps to Generate DPG
+
+1. **Configure tspconfig.yaml in spec repository**
+   
+   In your specs repository, update or create `tspconfig.yaml` to configure the TypeScript emitter. Replace `YOUR_SERVICE_DIRECTORY`, `YOUR_SERVICE_FOLDER` and `YOUR_PACKAGE_NAME` with your specific details.
+
+   ```yaml
+   parameters:
+     "service-dir":
+       default: "YOUR_SERVICE_DIRECTORY"
+
+   emit: ["@azure-tools/typespec-ts"]
+
+   options:
+     "@azure-tools/typespec-ts":
+       package-dir: "YOUR_SERVICE_FOLDER"
+       isModularLibrary: true
+       packageDetails:
+         name: YOUR_PACKAGE_NAME
+         description: "SHORT_DESCRIPTION"
+         version: "1.0.0-beta.1"
+   ```
```

Comment:
do we need to ask customers to

> **Note**: After the initial generation, add `generateMetadata: false` to avoid overwriting any manual updates to the generated package.json

Reply:
no, in the latest codegen we automatically calculate if we need to generate the metadata when generateMetadata is undefined.

the `false` means always not generating;
the `true` means always generating or overriding.